restaurants/views.py

```diff
@@ -23,16 +28,35 @@ def get_queryset(self):
         queryset = Restaurant.objects.all().order_by('id')
         return queryset
     
+
 class RestaurantDetailView(DetailView):
     model = Restaurant
     template_name = 'restaurants/detail.html'
     context_object_name = 'restaurant'
 
-
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["menu_items"] = self.object.menu_items.filter(is_deleted=False) 
         return context
     
-   
+
+class ReviewCreateView(LoginRequiredMixin, CreateView):
+    model = Review
+    form_class = ReviewForm
+    template_name = 'restaurants/review_form.html'
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        form.instance.restaurant_id = self.kwargs['id']
```

Comment:
This will raise unhandled error if some invalid restaurant id is provided in the URL. Before passing the data to the form, validate that the restaurant exists. Raise 404 if it does not.

You can do the following in dispatch() method after overriding.
`self.restaurant = get_object_or_404(Restaurant, pk=self.kwargs['id'])`